### PR TITLE
Automate nix flake update PRs

### LINKED ISSUE #4817: Automate nix flake update PRs
Our nix flake self-check CI started failing a few days ago due to `nixpkgs` being more than 30 days old. This is easily fixed by running `nix flake update` as was done to resolve the issue ([here](4789)). We should automate this and proactively create PRs to prevent CI from breaking again. See the "What to do" section on [this](https://github.com/fedimint/fedimint/actions/runs/8510775651) recent failed run for instructions on how to set this up. In short, we should use the [`update-flake-lock`](https://github.com/determinateSystems/update-flake-lock) GitHub action.

### PR DESCRIPTION
**Pull request by star-gazer111**
fixes  #4817

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -21,3 +21,5 @@ coverage:
     patch:
       default:
         target: 0 # we don't require any specific coverage for each PR
+
+
```
Comment (elsirion) [inline .codecov.yml:25]: Unrelated changes

File: .github/dependabot.yml [hunk 2]
```diff
@@ -9,3 +9,6 @@ updates:
   directory: "/"
   schedule:
     interval: daily
+              - name: Update Nix Flake Lock
+                uses: DeterminateSystems/update-flake-lock@v21
+
```
Comment (elsirion) [inline .github/dependabot.yml:14]: This doesn't look schema compliant https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

**Review by elsirion — CHANGES_REQUESTED** [review]

File: .codecov.yml (hunk 1 above)
Comment (star-gazer111) [reply .codecov.yml:25]: ya, those are unrelated, nothing changed there.

File: .github/dependabot.yml (hunk 2 above)
Comment (star-gazer111) [reply .github/dependabot.yml]: Ya sorry, I have made the changes now, please have a look.

Comment (elsirion) [reply .github/dependabot.yml]: You changed the indentation, I think you are confusing GitHub actions yaml files with dependabot.yml. These are two totally different config file types with different schemas (look at the link for available options).

File: .codecov.yml (hunk 1 above)
Comment (elsirion) [reply .codecov.yml:25]: Then they shouldn't be in the PR

File: .github/dependabot.yml [hunk 3]
```diff
@@ -9,3 +9,6 @@ updates:
   directory: "/"
   schedule:
     interval: daily
+- name: Update Nix Flake Lock
```
Comment (dpc) [inline .github/dependabot.yml:12]: These two lines:

image

;)

You're looking for `.github/workflows/ci-nix.yml`

Comment (dpc) [reply .github/dependabot.yml:12]: Actually, we probably want another workflow file, like `daily.yml`, but maybe let's do it once a week.

Comment (star-gazer111) [reply .github/dependabot.yml]: made the changes.

File: .codecov.yml (hunk 1 above)
Comment (star-gazer111) [reply .codecov.yml]: removed.

File: .github/workflows/ci-nix.yml [hunk 4]
```diff
@@ -34,7 +34,22 @@ jobs:
         with:
           fail-mode: true
           nixpkgs-keys: nixpkgs
-
+          
+  lockfile:
```
Comment (dpc) [inline .github/workflows/ci-nix.yml:38]: This flow runs on every PR, merge queue, etc. We probably want it just once a week or so.

Comment (star-gazer111) [reply .github/workflows/ci-nix.yml:38]: > This flow runs on every PR, merge queue, etc. We probably want it just once a week or so.

so you want the `update-flake-lock` flow to run weekly, correct?

Comment (tvolk131) [reply .github/workflows/ci-nix.yml:38]: Nix requires that it's updated at least every 30 days, so I think weekly seems reasonable

File: .github/workflows/ci-nix.yml [hunk 5]
```diff
@@ -34,7 +34,22 @@ jobs:
...
+    name: Update-flake-lock
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Nix
+      - uses: DeterminateSystems/nix-installer-action@main
+      - name: Update Nix Flake Lock
+      - uses: DeterminateSystems/update-flake-lock@v21
```
Comment (elsirion) [inline .github/workflows/ci-nix.yml:46]: I think there are a few `-` too many here. Afaik each one starts a new step.

**Comment by elsirion** [comment]
I think this would have been more efficient if DPC just did it himself. Too much review time wasted here already.
